src/QsCompiler/QirGeneration/DebugInfoManager.cs

```diff
@@ -170,46 +404,132 @@ private DebugInfoBuilder GetOrCreateDIBuilder(string sourcePath)
+        private void EmitLocation(Position absolutePosition, DILocalScope? localScope)
+        {
+            if (this.DebugFlag && localScope != null)
+            {
+                this.CurrentInstrBuilder.SetDebugLocation(ConvertToDebugPosition(absolutePosition), localScope);
+            }
+        }
+
+        /// <summary>
+        /// Converts from a 0-based <see cref="Position"/> to a 1-based <see cref="DebugPosition"/>.
+        /// </summary>
+        /// <param name="position">The 0-based <see cref="Position"/></param>
+        /// <return>The 1-based <see cref="DebugPosition"/></return>
+        private static DebugPosition ConvertToDebugPosition(Position position)
+        {
+            return DebugPosition.FromZeroBased(position.Line, position.Column);
+        }
 
-                // For now this is here to demonstrate a compilation unit being added, but eventually this will be called whenever a new file is encountered and not here.
-                // Get the source file path from an entry point.
-                string sourcePath = "";
-                if (!entryPoints.IsEmpty)
+        /// <summary>
+        /// Sums up the offsets from the stack of statement locations <see cref="StatementLocationStack"/>.
+        /// </summary>
+        /// <returns>The offset that the current stack of statements yields as a <see cref="Position"/>.</returns>
+        private Position TotalOffsetFromStatements()
+        {
+            Position offset = Position.Zero;
+
+            foreach (QsLocation? loc in this.StatementLocationStack)
+            {
+                if (loc != null)
                 {
-                    if (this.sharedState.TryGetGlobalCallable(entryPoints[0], out QsCallable? entry))
-                    {
-                        sourcePath = entry.Source.CodeFile;
-                    }
-                    else
-                    {
-                        throw new Exception("Entry point not found");
-                    }
+                    offset += loc.Offset;
+                    return offset;
                 }
-                else
+            }
+
+            return offset;
+        }
+
+        /// <summary>
+        /// Sums up the offsets from the stack of expression ranges <see cref="ExpressionRangeStack"/>.
+        /// </summary>
+        /// <returns>The offset that the current stack of expression yields as a <see cref="Position"/>.</returns>
+        private Position TotalOffsetFromExpressions()
+        {
+            Position offset = Position.Zero;
+
+            foreach (DataTypes.Range? range in this.ExpressionRangeStack)
+            {
+                if (range != null)
                 {
-                    throw new Exception("No entry point found in the source code");
+                    offset += range.Start;
                 }
-
-                this.GetOrCreateDIBuilder(sourcePath);
             }
+
+            return offset;
         }
+
+// SECTION: Private getters
```

Comment:
I think this private getters are not needed.

Reply:
I included them in case we ended up needing to perform some logic to when deciding the versions (like check the target architecture etc.) But changing it if/when we need to do that shouldn't be difficult, so should I remove them and return the values directly?